### PR TITLE
Add support for upstream MCP registry format

### PR DESCRIPTION
**Pull request by JAORMX**
## Problem

ToolHive currently uses its own registry format, which creates barriers for ecosystem interoperability. The MCP community has developed a standardized \`server.json\` format that other tools use, but ToolHive cannot consume or produce this format.

## Solution

This PR adds bidirectional conversion between ToolHive's registry format and the upstream MCP community format. This enables ToolHive to participate in the broader MCP ecosystem while preserving its advanced container-focused features.

### New Go Packages
- **\`pkg/registry/upstream.go\`**: Structs matching the upstream \`server.json\` schema
- **\`pkg/registry/upstream_conversion.go\`**: Conversion functions in both directions
- **\`pkg/registry/upstream_conversion_test.go\`**: Comprehensive test coverage

### Format Mapping Strategy

**Upstream → ToolHive:**
- \`docker\` packages → Direct image references (\`nginx:latest\`)
- \`npm\` packages → NPX protocol schemes (\`npx://package@version\`)
- \`pypi\` packages → UVX protocol schemes (\`uvx://package@version\`)
- \`remotes\` → \`RemoteServerMetadata\`

**ToolHive → Upstream:**
- Container servers → \`docker\` packages
- Protocol schemes → Appropriate registry packages
- ToolHive-specific metadata → \`x-publisher.build_info.toolhive\` extension

### Extension Mechanism

ToolHive's advanced features (permissions, tiers, provenance) are preserved using the \`x-publisher\` extension field, following community conventions for vendor-specific metadata.

## Design Decisions

1. **Bidirectional conversion**: Enables both consuming upstream formats and contributing back to community registries
2. **Extension-based approach**: Preserves ToolHive features without breaking upstream compatibility
3. **Protocol scheme mapping**: Leverages ToolHive's existing \`npx://\`/\`uvx://\` support for non-Docker packages
4. **Zero breaking changes**: Existing ToolHive registry format remains unchanged

## Review Focus Areas

- **Conversion accuracy**: Check the mapping logic in \`upstream_conversion.go\`
- **Extension design**: Review the \`ToolhivePublisherExtension\` struct for completeness
- **Error handling**: Verify graceful handling of malformed or incomplete upstream data
- **Test coverage**: Ensure edge cases are covered in the test suite

## Future Work

This foundation enables future features like:
- CLI commands for importing/exporting upstream formats
- Automatic synchronization with community registries
- Validation tools for upstream format compliance

See \`docs/proposals/upstream-mcp-registry-format-support.md\` for detailed technical design.

### CODE REVIEW

File: pkg/registry/upstream_conversion.go [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+					Version:              extractImageTag(imageServer.Image),
+					PackageArguments:     convertArgsToUpstream(imageServer.Args),
+					EnvironmentVariables: convertEnvVarsToUpstream(imageServer.EnvVars),
+				},
+			}
+		}
+	}
+
+	// Add toolhive extension to x-publisher
+	upstream.XPublisher = &UpstreamPublisher{
```
Comment (rdimitrov) [inline pkg/registry/upstream_conversion.go:198]: I wonder if we should have those nested/namespaced, i.e. 
```json
{
...,
  "x-publisher": {
    "x-com.github.stacklok.toolhive": {...}, /* or just toolhive if we want to start simple */
  }
}
```

**Review by rdimitrov — APPROVED** [review]
